### ISSUE TITLE
Difficulty defining a "one-sided" belongsTo relationship

### ISSUE THREAD

**Issue by blimmer** [samselikoff/ember-cli-mirage#681]
First off, thank you for taking the time to help me with all of these questions. I'm really struggling to wrap my head around the new model/serialization structures.

I'm trying to set up this sort of behavior:

I have two models: `User` and `Metric`. `Metric` belongsTo a `User`, but `User` doesn't include the metrics_id. For instance, the payload I'm wanting Mirage to send down is something like this:

``` javascript
// GET /user/1

{
  "user": {
    "id": 1,
    "name": "Foo"
  },
  "metrics": [
    {
      "id": "2016-04-06T16:07:58.166Z",
      "user_id": 1
    }
  ]
}
```

Because of some complexities around the generation of the metric id, it's desirable for the server team to not include the `metrics_id` on the `user` payload.

Is there a way to do this?

**Comment by samselikoff** [comment]
I believe right now the `include` option on serializers both side-loads models, and adds their `id` to the parent, and we should definitely add an `embed: :ids` option that's separate from `include`. 

For now, you can always [override serialize](http://www.ember-cli-mirage.com/docs/v0.2.0-beta.7/serializers/#serializeresponse-request) on the model-specific serializer and munge the data:

```js
serialize(object, request) {
  let json = Serializer.prototype.serialize.apply(this, arguments);

  delete json.metrics_id;

  return json;
}
```

**Comment by blimmer** [comment]
I'm having trouble even getting the basic two-way belongsTo set up.

I set up this sample showing what I'm expecting to work: https://github.com/samselikoff/ember-cli-mirage/issues/681

this is the result:
```json
{
  "user": {
    "id": "1",
    "metric_id": null
  }
}
```

Can you see what I'm missing?

**Comment by samselikoff** [comment]
you linked to this issue, did you mean to link somewhere else?

**Comment by blimmer** [comment]
Crud sorry

https://github.com/blimmer/ember-cli-example-app-for-github/tree/ember-cli-mirage/681?files=1

On Thu, Apr 7, 2016 at 17:53 Sam Selikoff <notifications@github.com> wrote:

> you linked to this issue, did you mean to link somewhere else?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/samselikoff/ember-cli-mirage/issues/681#issuecomment-207141745>
>
-- 
[sent from my mobile]

**Comment by samselikoff** [comment]
- one-to-one relationships aren't _really_ supported yet. You can fake it at the serializer layer. I would make one side `hasMany` then munge the data in `serializer#serialize`
- your `UserSerializer` says `include: ['metrics']` but the user doesn't have any relationships under the key of `metrics`... it has a `metric` (singular). So the includes here are pointing to named associations.
- Even if you correct `metrics` to `metric` it still won't work. Let me explain why. Right now since 1-1 relationships aren't supported, `belongsTo` assumes the child owns the `fk`. So, when you put `metric: belongsTo()` in the User model, you're telling Mirage that a `user` has a `metricId` foreign key, and that's how it looks up the attribute. But in the scenario, you create a `metric` with a `userId`. So when the serializer asks the user if it's associated `metric` model exists, it gets null.

    You can either switch the scenario around so you first create the `metric` and then create the user passing in the `{ metric }`, or turn `user.metrics` into a `hasMany` - but then you'd have to do some munging at the serializer layer.

Does that clear things up? I know this is frustrating/broken right now, I have `hasOne` built at the orm layer but I need to update the serializer layer to support it.

**Comment by workmanw** [comment]
👍 I too ran into this issue.

**Comment by workmanw** [comment]
At first I thought I just needed a way to override `getForeignKey` on the `belongsTo` association. So I tried to extend `BelongsTo`, but unfortunately it turned out more challenging. In our API, relationship fields have the same name as the key (they don't have an alternate name for the ID). So for example:

```
{
  blog_posts: {
    name: '...',
    author: 12
  },
  authors: [
    { id: 12, first_name: 'Hello, last_name: 'World' }
  ]
}
```

Unfortunately when I tried to override `belongsTo`  I get the following error: `TypeError: Cannot redefine property: champion`. This occurs because of the implementation of `belongsTo` just redefines the properties: https://github.com/samselikoff/ember-cli-mirage/blob/master/addon/orm/associations/belongs-to.js#L19-L30

Anyways, I know all of this stuff is really complicated because there are hundreds of variations our APIs have. I'm not expecting a response or workaround, just trying to provide the use case so you have it.
